services/entitlement/service/src/graphql/plugins/entitlement-endpoint/catalog-service-communication.ts

```diff
@@ -89,8 +89,7 @@ const getCatalogResponseVideo = (
   if (
     !videos[0].isProtected ||
     !videos[0].videoStreams ||
-    videos[0].videoStreams.nodes.length === 0 || // A protected video will always have a video_streams record
-    videos[0].videoStreams.nodes
+    (videos[0].videoStreams?.nodes ?? []) // A protected video will always have at least one stream with non-empty Key ID
```

Comment:
I was trying to combine length and null/undefined checks. Refactored this part of the code a bit so it would still include the length check, not include the `?? []`, but would still make existing tests pass.